src/index/hnsw.cpp

```diff
@@ -273,9 +277,13 @@ HNSW::knn_search(const DatasetPtr& query,
                 return this->alg_hnsw_->getDistanceByLabel(label, vector);
             };
             conjugate_graph_->EnhanceResult(results, func);
+            k = original_k;
         }
 
         // return result
+        while (results.size() > k) {
+            results.pop();
+        }
```

Comment:
comparing with original search process, here only move partial pop operations from hnswalg to hnsw. So it wouldn't affect the search speed.